### ISSUE TITLE
[Bug]cannot enable notification for website

### ISSUE THREAD

**Issue by ale82to** [fork-maintainers/iceraven-browser#87]
when I try to enable push notification after the prompt for allowing notification, Twitter website tells me that my browser is not supporting push notification in fenix i ve got no problems at all

**Comment by interfect** [comment]
This could be a result of all the Google/Firebase/Mozilla telemetry stuff we ripped out. Push notifications on the browser side are implemented using backend servers that the browser talks to, and I think Firebase might have backed the push system in Android Firefox.

**Comment by ale82to** [comment]
yes very likely I hope this will be fixed for me is an eseential feature
thabks

**Comment by ghost** [comment]
I contend we ought not to bring any sort of proprietary backend back into the browser. For me, notifications were (are?) working fine (at least for invidious)

**Comment by interfect** [comment]
Normal, page-initiated notifications should work fine, but server-initiated *push* notifications don't work. https://webpushdemo.azurewebsites.net/ can prompt for permission in Mozilla Firefox from Google Play, but not in Iceweasel.

It might be that you need not only the Firebase dependency but also Mozilla's API key to make it work the way they do. We need to check if a from-source build of Mozilla Firefox, without the Mozilla keys, works or not.

If it does work, we can think about pulling in the Firebase SDK again. I'm not even sure that it *is* closed source, it just has a SaaS backend and is not available in F-Droid's approved Maven sources. It might be possible to add it as a source dependency for F-Droid, or replace it with an open implementation that talks to the same backend.

If we need API keys, we'd either have to get our own, check them in, and repeat every time someone steals them, or else find some other way to implement web push not involving Firebase, maybe with our own backend. Either way might cost money. I suppose we could try and add some kind of IAP to pay for it, but making that work without Google Play would be interesting.

Maybe we can add a setting where you fill in a backend?

**Comment by interfect** [comment]
The Firebase push SDK is open source, so we could probably copy-paste it if it came to that, but I think it only works if Google Play Services is installed, and if you have an API key.

https://bubu1.eu/openpush/ is an open front-end and back-end pair, but someone needs to run the backend somewhere.

**Comment by bbigam** [comment]
I don't like browser push notifications myself; that said, I'd support bringing them back (since they can easily be turned off) if and only if we can do so in a way that is totally FOSS with no proprietary dependencies.

**Comment by Snuupy** [comment]
Looks like openpush has been discontinued and https://unifiedpush.org/ is the new method of implementing push notifications. Does anyone have any pointers on how this might be implemented? I see the following relevant files:

- https://github.com/fork-maintainers/iceraven-browser/blob/fork/app/src/main/java/org/mozilla/fenix/components/Push.kt 
- https://github.com/mozilla-mobile/fenix/blob/51cec928b2c9d56b9fc380e0bbe44fd2d74025ab/app/src/main/java/org/mozilla/fenix/push/WebPushEngineIntegration.kt
- https://github.com/fork-maintainers/iceraven-browser/blob/fork/app/src/main/java/org/mozilla/fenix/FenixApplication.kt#L359

but I'm not sure if it can be used as a template on implementing unifiedpush.
